### PR TITLE
Update error message of `ErrorInDataBindingGroup` to include participating identifiers

### PR DESCRIPTION
**Pull request by LiamGoodacre**
Should help with #2431 

Given the following code:
```purescript
data Foo = MakeFoo Bar
data Bar = MakeBar (Foo Int)
```

The error message is now:
```
  Could not match kind

    * -> *

  with kind

    *

  in data binding group Bar, Foo
```

### CODE REVIEW

File: src/Language/PureScript/TypeChecker.hs [hunk 1]
```diff
@@ -205,9 +206,10 @@ typeCheckAll moduleName _ = traverse go
       addDataType moduleName dtype name args' dctors ctorKind
     return $ DataDeclaration dtype name args dctors
   go (d@(DataBindingGroupDeclaration tys)) = do
-    warnAndRethrow (addHint ErrorInDataBindingGroup) $ do
-      let syns = mapMaybe toTypeSynonym tys
-      let dataDecls = mapMaybe toDataDecl tys
+    let syns = mapMaybe toTypeSynonym tys
+    let dataDecls = mapMaybe toDataDecl tys
```
Comment (paf31) [inline src/Language/PureScript/TypeChecker.hs]: Nitpick: can we remove the unnecessary `let`s here?

File: src/Language/PureScript/TypeChecker.hs [hunk 2]
```diff
@@ -205,9 +206,10 @@ typeCheckAll moduleName _ = traverse go
...
+    let bindingGroupNames = (syns^..traverse._1) ++ (dataDecls^..traverse._2)
```
Comment (paf31) [inline src/Language/PureScript/TypeChecker.hs]: I think you could use `both` here. Or just `map fst` and `map snd` to keep things simple, perhaps?

Also, does this need a `nub`?

**Comment by paf31** [comment]
👍  Great, thanks!

Closes #2431

File: src/Language/PureScript/TypeChecker.hs (hunk 2 above)
Comment (LiamGoodacre) [inline src/Language/PureScript/TypeChecker.hs]: I don't think I can use `map snd` as `dataDecls` elements are triples.
I can switch to `map (\(_, n, _) -> n) dataDecls` if you prefer?  There is similar code on the next few lines.
Using `nub` sounds good.

Comment (LiamGoodacre) [inline src/Language/PureScript/TypeChecker.hs]: sorry 4-tuples not triples

Comment (paf31) [inline src/Language/PureScript/TypeChecker.hs]: Ah, gotcha. Then this is fine as-is.

**Pull request merged**